geomstats/geometry/pullback_metric.py

```diff
@@ -138,20 +139,24 @@ def immersion_a(x):
             )
             if self.dim == 1 and hessian_a.ndim > 2:
                 hessian_a = gs.squeeze(hessian_a, axis=-1)
+            elif self.dim != 1 and hessian_a.ndim > 2:
+                hessian_a = gs.squeeze(hessian_a)
+
             hessian_aij.append(hessian_a)
 
             jacobian_a = gs.squeeze(jacobian_a, axis=0)
             if len(jacobian_a.shape) == 0:
                 jacobian_a = gs.to_ndarray(jacobian_a, to_ndim=1)
+            elif self.dim != 1 and jacobian_a.ndim > 1:
+                jacobian_a = gs.squeeze(jacobian_a)
             jacobian_ai.append(jacobian_a)
 
         hessian_aij = gs.stack(hessian_aij, axis=0)
         jacobian_ai = gs.stack(jacobian_ai, axis=0)
         inner_prod_deriv_mat = gs.einsum(
-            "aki,aj->kij", hessian_aij, jacobian_ai
-        ) + gs.einsum("akj,ai->kij", hessian_aij, jacobian_ai)
+            "aki,aj->ijk", hessian_aij, jacobian_ai
```

Comment:
is it working with multiple points?

Reply:
It will with the PR on the pulbackmetric, I'll address it there.